_posts/2020-06-18-a-sustainable-nuget-marketplace-will-have-to-compete-with-the-nuget-gallery.markdown

```diff
@@ -0,0 +1,376 @@
+
+While not a key component of this NuGet marketplace, it may be a good incentive for consumers to flock here
+and find those services.
+
+### For producers and consumers: reporting
+
+As a producer, I will want to know who is using packages. How many downloads are there? What's the revenue our project is making?
+
+As a consumer, I want to know who I'm purchasing from. I want invoices, reports on who of my staff are using which licenses, ...
+
+### For producers and consumers: a solid platform
+
+This NuGet marketplace will start off as a novel concept and will have to win hearts with producers and consumers.
+Over time, it will have to build and maintain trust. As first impressions matter, it will have to be built
+as a solid platform, from a technical point of view, UX point of view, and the aformentioned legal and business
+services. A Minimum Viable Product (MVP) approach may help in building this solid platform with laser focus.
+
+Speaking from experience building out [MyGet](https://www.myget.org/) in the past, this NuGet marketplace
+will get hammered by users. NuGet attempts to retrieve every package installed from every configured feed,
+so expect to serve up a lot of 404 not found responses. These, too, will require a solid platform.
+
+A solid platform also means ease of use, and ease of discovery. One thing that could help here is providing
+credential providers for [Visual Studio](https://docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-credential-providers-for-visual-studio) and [JetBrains Rider](https://blog.jetbrains.com/dotnet/2018/03/06/credential-providers-private-nuget-feeds-rider-2018-1-eap/).
+By extension, maybe also for the .NET command line tools. These will prompt the user to authenticate, and are an in-IDE
+experience where payment detail updates and all that can be arranged. Discovery is key, and bringing
+part of what this NuGet marketplace will offer in its web application into the IDE may help.
+
+## A marketplace has two sides - getting traction
+
+Marketplaces have two sides, producers and consumers. If a NuGet marketplace comes about, it will have
+to accommodate both, and probably subsidize one side for a while until traction takes over.
+
+So how can this NuGet marketplace get traction? There are a few things that can be considered.
+
+### Getting consumers on board
+
+The marketplace will need consumers, and folks with credit cards and purchase orders. What's their incentive to look
+at this solution? I think the answer is in all of the topics above, where ease of use, legal and purchasing,
+single point of contact come to mind.
```

Comment:
For big companies, budgeting is part of the issue - if Sdkbin is the merchant of record, development teams can set their budget against Sdkbin and then allocate it across the annual licenses / new licenses they use without having to onboard additional vendors. Cuts down on the bureaucratic overhead of adopting new solutions tremendously.